### PR TITLE
Add some factories and specs for models

### PR DESCRIPTION
**Pull request by smoothcontract**
### Context
I'll be using these factories shortly for setting up some test data fixtures, but I think this is already useful as a standalone PR.

### Changes proposed in this pull request
This adds some factory bot factories for the key models using Faker to populate attributes. Also added some very simplistic tests to debug the has many relationships - all working nicely now.

### Guidance to review

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
 class Category < ApplicationRecord
+  has_many :job_profile_categories
```
Comment (cwrw) [inline app/models/category.rb]: could we add the `inverse_of` for all the associations?

File: app/models/job_profile.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class JobProfile < ApplicationRecord
+  has_many :job_profile_categories
```
Comment (cwrw) [inline app/models/job_profile.rb]: If a `JobProfileCategory` is deleted, do we want to persist Job Profiles?

File: app/models/category.rb (hunk 1 above)
Comment (smoothcontract) [reply app/models/category.rb]: Hadn’t come across that before so googled - seems in Rails > 4 those are automatically determined 🤘

https://makandracards.com/makandra/12331-rails-when-to-use-inverse_of-in-has_many-has_one-or-belongs_to-associations

File: spec/models/category_spec.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
+
+RSpec.describe Category do
+  let(:category) { FactoryBot.build_stubbed(:category) }
```
Comment (cwrw) [inline spec/models/category_spec.rb]: you can call `build_stubbed` directly without using `FactoryBot` as so: `let(:category) {  build_stubbed(:category) }`

File: app/models/category.rb (hunk 1 above)
Comment (cwrw) [reply app/models/category.rb]: It seems for the `through` relation they aren't:
```
Below there are some known options where a automatically determination of the inverse is not working:

:conditions, :through ...
```
Could we add them there?

File: app/services/seed_service.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  class << self
+    def seed
+      only_development!
+      delete_all
+      seed_categories
+      seed_skills
+      seed_job_profiles
+    end
+
+    private
```
Comment (cwrw) [inline app/services/seed_service.rb]: this is odd, according the linter:
```
Linting ruby...
app/services/seed_service.rb:11:5: C: [Corrected] Layout/AccessModifierIndentation: Outdent access modifiers like private. (https://github.com/rubocop-hq/ruby-style-guide#indent-public-private-protected)
    private
    ^^^^^^^
```

Comment (cwrw) [reply app/services/seed_service.rb]: I would have assumed what you did was correct 🤷‍♂

Comment (smoothcontract) [reply app/services/seed_service.rb]: Hmm.. maybe private class methods are not a thing. I'll refactor to move some of that stuff to instance methods.

**Comment by smoothcontract** [comment]
> I don't usually test associations as they are part of the framework. When we add implementation we will be testing all these associations and behaviours fully, so I would recommend we remove those for now. What are your thoughts?

I'd agree with this, but having the specs within this PR at least has been useful to get the relationships defined correctly. Once we have some more realistic test coverage (i.e. anything that actually exercises the models) we can remove the association tests as part of the relevant PR.

File: app/models/job_profile.rb (hunk 2 above)
Comment (smoothcontract) [reply app/models/job_profile.rb]: Good question - I'd assume yes for now. I may revise this when working further on the web scraper.

**Pull request merged**